### PR TITLE
Add 30s timeout to pytest run

### PR DESCRIPTION
**Pull request by tjguk**
Add a 30s timeout to the pytest run in the hope that -- if tests hang, they'll fail faster!

### CODE REVIEW

**Comment by tjguk** [comment]
Well that doesn't seem to have helped at all :) I can see the 30s timeout noted on each of the test runs, but 3.7 on Ubuntu latest has been running for 20+ minutes with no results and no timeout!

**Comment by tjguk** [comment]
I think this is about as good as it's going to get for the moment. In sum, it looks like something in test_app is causing the hangs. For now, I've left `test_app:test_run` skipped as that seems to give the best outcome

File: tests/interface/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# ~ import pytest
+
+# ~ pytestmark = pytest.mark.skip("Skipping for hangs")
```
Comment (carlosperate) [inline tests/interface/__init__.py:3]: Should we delete all these commented out skips or do you still need them for more testing?

Comment (tjguk) [reply tests/interface/__init__.py:3]: Those can definitely go -- I'll raise a small PR. (Sorry; I've gone backwards and forwards so much over the code, trying to narrow down the test failures that I forgot about them..)

**Pull request merged**